quinn-proto/src/connection.rs

```diff
@@ -2223,6 +2223,14 @@ impl Connection {
         stream: StreamId,
         data: &[u8],
     ) -> Result<usize, WriteError> {
+        if self.state.as_ref().unwrap().is_closed() {
+            return Err(WriteError::Blocked);
+        }
+        assert!(stream.directionality() == Directionality::Bi || stream.initiator() == self.side);
+        if self.blocked() {
+            self.blocked_streams.insert(stream);
+            return Err(WriteError::Blocked);
+        }
```

Comment:
This partially defeats the debug logging just below, preventing some cases of blocked writes from being traced. What's the motivation?

Reply:
I think code is often easier to follow if error conditions and such are handled first, leading to early returns (or continue/break), then the actual logic follows. I've adjusted the debug logging now, so that it's actually more specific. What do you think?

Reply:
Looks good! Wouldn't hurt to log the affected stream ID too.

Reply:
Done.